src/pineko/cli/check.py

```diff
@@ -36,54 +38,46 @@ def sub_compatibility(grid_path, operator_path, xif):
         rich.print("[red]Error:[/]", e)
 
 
+class Scale(Enum):
+    """Auxiliary class to list the possible scale variations."""
+
+    REN = auto()
+    FACT = auto()
+
+
 @subcommand.command("scvar")
 @click.argument("grid_path", metavar="PINEAPPL", type=click.Path(exists=True))
 @click.argument(
     "scale",
     metavar="SCALE",
-    type=click.Choice(["ren", "fact"]),
+    type=click.Choice(list(map(lambda x: x.name.upper(), Scale)), case_sensitive=False),
 )
 @click.argument("max_as_order", metavar="AS_ORDER", type=int)
 @click.argument("max_al_order", metavar="AL_ORDER", type=int)
 def sub_scvar(grid_path, scale, max_as_order, max_al_order):
     """Check if PineAPPL grid contains requested scale variations for the requested order."""
     grid = pineappl.grid.Grid.read(grid_path)
     grid.optimize()
-    if scale == "ren":
-        is_ren_as, is_ren_al = check.contains_ren(grid, max_as_order, max_al_order)
-        if is_ren_as:
-            rich.print(
-                "[green]Success:[/] grids contain renormalization scale variations for as"
-            )
-        else:
-            rich.print(
-                "[red]Error:[/] grids do not contain renormalization scale variations for as"
-            )
-        if is_ren_al:
-            rich.print(
-                "[green]Success:[/] grids contain renormalization scale variations for al"
-            )
-        else:
-            rich.print(
-                "[red]Error:[/] grids do not contain renormalization scale variations for al"
-            )
-    elif scale == "fact":
-        is_fact_as, is_fact_al = check.contains_fact(grid, max_as_order, max_al_order)
-        if is_fact_as:
-            rich.print(
-                "[green]Success:[/] grids contain factorization scale variations for as"
-            )
-        else:
-            rich.print(
-                "[red]Error:[/] grids do not contain factorization scale variations for as"
-            )
-        if is_fact_al:
-            rich.print(
-                "[green]Success:[/] grids contain factorization scale variations for al"
-            )
+    success = "[green]Success:[/] grids contain"
+    error = "[red]Error:[/] grids do not contain"
+    scale_variations = {
+        Scale.REN.name: " renormalization scale variations",
+        Scale.FACT.name: " factorization scale variations",
+    }
+    if scale not in scale_variations.keys():
+        raise ValueError("Scale variation to check can be one between ren and fact")
+    sv = scale_variations[scale]
+    funcs = {Scale.REN.name: check.contains_ren, Scale.FACT.name: check.contains_fact}
```

Comment:
Also this one, you can assign to the value of `Scale` (just make it a tuple, or a dedicate object).

E.g.:
```py
class Scale(Enum):
    REN = ("renormalization scale variations", check.contains_ren)
    ...
```

There is also the intermediate option, you can make it a [`namedtuple`](https://docs.python.org/3/library/collections.html?highlight=namedtuple#collections.namedtuple):

```py
from collections import namedtuple

ScaleValue = namedtuple('ScaleValue', ['descr', 'check'])

class Scale(Enum):
    REN = ScaleValue("renormalization scale variations", check.contains_ren)
    ...
    
Scale[myscale].value.check
```

Reply:
> Named tuples assign meaning to each position in a tuple and allow for more readable, self-documenting code.